lib/db/schema/patch-095-096.sql

```diff
@@ -0,0 +1,298 @@
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('95');
+
+-- This migration adds an optional `refreshTokenId` column to the `devices` table,
+-- allowing OAuth clients to participate in the sync device ecosystem.
+
+-- First, in order for the migration to apply cleanly via migration tooling,
+-- we need to drop `FOREIGN KEY` constraints on the `devices` table. There's
+-- one instance of this in the old `deviceCapabilities` table which is no
+-- longer used and should be safe to drop entirely.
+
+DROP TABLE IF EXISTS deviceCapabilities;
+
+-- And there are two foreign keys on the `deviceCommands` table, so we might
+-- as well remove both while we're here.  The first links to the `deviceCommandIdentifiers`
+-- table, from which we never delete and so which will not be affected by removal.
+-- The second links to `devices` so we need to replace the `ON DELETE CASCADE`
+-- with manual deletion in a couple of stored procedures.
+
+ALTER TABLE deviceCommands
+  DROP FOREIGN KEY deviceCommands_ibfk_1,
+  DROP FOREIGN KEY deviceCommands_ibfk_2,
+ALGORITHM = INPLACE, LOCK = NONE;
```

Comment:
Here's a naive question because DB's aren't really my strong point. For migrations, could we drop the FK constraint for the migration, do the migration, and re-add the FK constraint?

Another naive question, if we remove the FK constraints and the cascading deletes, then there is a window where the cascading deletes are not happening and the new stored procedures are not yet being called. Are we going to have orphaned rows in deviceCommands? 

cc @jrgm

Reply:
> could we drop the FK constraint for the migration, do the migration, and re-add the FK constraint?

We could, but IIURC adding th FK constraint back in is a costly operation, because it has to check every row in the table to ensure it doesn't violate the constraint.  I interpret @jrgm's comment in https://github.com/mozilla/fxa-auth-db-mysql/issues/485#issue-416482326 about now being able to use `rebuild_constraints`, as applying to the expensive nature of this process on a large table.

> if we remove the FK constraints and the cascading deletes, then there is a window where the
> cascading deletes are not happening and the new stored procedures are not yet being called.
> Are we going to have orphaned rows in deviceCommands?

Good point, I think you're correct there's a chance of leaving some.

We could split this migration in two, making one release to add the new SPs without actually removing the FK constraints, and then another release with the rest of the migration.

Alternately, we could do some out-of-band scan and cleanup of orphaned rows after the migration.

(Or, I suppose, we could just accept that there will be orphaned rows and not worry too much about it).